pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -235,25 +238,25 @@ func (f *ring0Factory) JSONEncoder() runtime.Encoder {
 	return legacyscheme.Codecs.LegacyCodec(legacyscheme.Registry.EnabledVersions()...)
 }
 
-func (f *ring0Factory) UpdatePodSpecForObject(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error) {
+func (f *ring0Factory) UpdatePodSpecForObject(obj runtime.Object, fn func(*v1.PodSpec) error) (bool, error) {
 	// TODO: replace with a swagger schema based approach (identify pod template via schema introspection)
 	switch t := obj.(type) {
-	case *api.Pod:
+	case *v1.Pod:
 		return true, fn(&t.Spec)
-	case *api.ReplicationController:
+	case *v1.ReplicationController:
 		if t.Spec.Template == nil {
-			t.Spec.Template = &api.PodTemplateSpec{}
+			t.Spec.Template = &v1.PodTemplateSpec{}
 		}
 		return true, fn(&t.Spec.Template.Spec)
-	case *extensions.Deployment:
+	case *appsv1beta1.Deployment:
```

Comment:
deployment exists in extensions/v1beta1, apps/v1beta1, and apps/v1beta2... need to select all three